pkg/storage/storage.go

```diff
@@ -1042,6 +1055,19 @@ func (is *ImageStore) DeleteBlob(repo string, digest string) error {
 		}
 	}
 
+	f, err := os.Open(blobPath)
+	if err != nil {
+		is.log.Error().Err(err).Str("blob", src).Msg("failed to open blob")
+		return err
+	}
+	// unset immutable attribute
+	err = chattr.UnsetAttr(f, chattr.FS_IMMUTABLE_FL)
+	if err != nil {
+		is.log.Error().Err(err).Str("blobPath", blobPath).Msg("Unable to remove immutable attribute from blob")
+		return err
```

Comment:
I think we leak f here, right? We need a defer f.Close() above

Reply:
Yes, that's right, thanks, I will update it.